xbmc/addons/AddonRepos.h

```diff
@@ -0,0 +1,188 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "AddonDatabase.h"
+
+#include <map>
+#include <vector>
+
+namespace ADDON
+{
+/**
+ * Struct - RepoInfo
+ */
+struct RepoInfo
+{
+  std::string m_repoId;
+  std::string m_origin;
+};
+
+class CAddonMgr;
+class CRepository;
+class IAddon;
+
+/**
+ * Class - CAddonRepos
+ */
+class CAddonRepos
+{
+public:
+  CAddonRepos(const CAddonMgr& addonMgr) : m_addonMgr(addonMgr){}; // ctor
+
+  /*!
+   * \brief Load the map of all available addon versions in any installed repository
+   * \param database reference to the database to load addons from
+   */
+  void LoadAddonsFromDatabase(const CAddonDatabase& database);
+
+  /*!
+   * \brief Load the map of all available versions of an addonId in any installed repository
+   * \param database reference to the database to load addons from
+   * \param addonId the addon id we want to retrieve versions for
+   */
+  void LoadAddonsFromDatabase(const CAddonDatabase& database, const std::string& addonId);
+
+  /*!
+   * \brief Build the list of addons to be updated depending on defined rules
+   * \param installed vector of all addons installed on the system that are
+   *        checked for an update
+   * \param [out] updates list of addon versions that are going to be installed
+   */
+  void BuildUpdateList(const std::vector<std::shared_ptr<IAddon>>& installed,
+                       std::vector<std::shared_ptr<IAddon>>& updates) const;
+
+  /*!
+   * \brief Checks if the origin-repository of a given addon is defined as official repo
+   *        but does not check the origin path (e.g. https://mirrors.kodi.tv ...)
+   * \param addon pointer to addon to be checked
+   * \return true if the repository id of a given addon is defined as official
+   */
+  static bool IsFromOfficialRepo(const std::shared_ptr<IAddon>& addon);
```

Comment:
**As note:**
About here on this request and also on older already present functions in maneger I look to replace `IAddon` with `CAddonInfo` where possible.

So that `IAddon` only be becomes created if related addon becomes used for work. This class do a much of works where mostly never needed (whole addon creation).

For now OK :+1:

Reply:
so is `CAddonInfo` an equal replacement for `IAddon` ? then i'd sure replace it.

Reply:
Yes if caller need only informations about his addon.xml content. 
Only not so easy to make quickly on new parts as the most callers still use whole CAddon (IAddon) on other places.

Reply:
attempt to bulk change `IAddon` -> `CAddonInfo` failed :(
so we should consider doing this successively in a follow up as it doesn't impact functionality

Reply:
Thats why I mean before "For now OK" and only as note before :smile:, this boths are a bit different and is not easily to use.

Has planned for **me** to improve this, but becomes heavier and your way complete OK for now.

Reply:
oh yeah, thanks 😊